### PR TITLE
feat: Dropstab v1.1

### PR DESCRIPTION
**Pull request by Motion-Up**
## Description

# DropsTab Dropstab Crypto Toolbox

### Search

Dropstab Crypto Toolbox displays the price, market capitalization, and rank of any cryptocurrency, and also allows quick access to the project's page on Dropstab.com or its X (Twitter) profile.

DropsTab

### Wallets

Dropstab Crypto Toolbox allows you to instantly find and view any crypto address in the blockchain explorer.
DropsTab

### Ethereum Gas Tracker

Get the latest gas prices for Ethereum. This feature provides real-time information on the current gas prices required for transactions on the Ethereum network.

DropsTab

### Macro View

Get the latest macroeconomic data. This feature provides a comprehensive overview of key macroeconomic indicators such as stock indices, commodities, and other financial instruments.

DropsTab

### Gainers & Losers

Get the latest gainers and losers. This feature displays the top-performing and worst-performing cryptocurrencies over the last 24 hours.

DropsTab

### CEX Pairs Search

Search for pairs on CEX. This feature allows users to search for trading pairs on centralized exchanges (CEX).

DropsTab

_Made with ❤ for the Decentralized World._

### Added

- Ethereum Gas Tracker
- Macro View
- Gainers & Losers
- CEX Pairs Search

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by Motion-Up** [comment]
@pernielsentikaer
Good afternoon! My update is ready.

**Comment by Motion-Up** [comment]
This pull request includes significant updates to the Dropstab Crypto Toolbox extension, introducing new features and enhancing existing ones. The most important changes include the addition of new features like Ethereum Gas Tracker, Macro View, Gainers & Losers, and CEX Pairs Search, along with updates to the local storage handling and package dependencies.

### New Features:
* **Added Ethereum Gas Tracker, Macro View, Gainers & Losers, and CEX Pairs Search**: These new features provide real-time gas prices for Ethereum, macroeconomic data, top-performing and worst-performing cryptocurrencies, and search functionality for trading pairs on centralized exchanges. (`extensions/dropstab/CHANGELOG.md`, `extensions/dropstab/README.md`, `extensions/dropstab/package.json`, `extensions/dropstab/src/gas-tracker.tsx`, `extensions/dropstab/src/gainers-and-losers.tsx`, `extensions/dropstab/src/exchange-search.tsx`) [[1]](diffhunk://#diff-b87f3c185a378b4c679ce5aedcb7810e641dda001ffe6395bc4199855afb59fbR4-R12) [[2]](diffhunk://#diff-fe0bcafb44232aa92569c7ed46acb031bbcd110517e51cb0761ea34e8e8a8d8bR4-R43) [[3]](diffhunk://#diff-fe0bcafb44232aa92569c7ed46acb031bbcd110517e51cb0761ea34e8e8a8d8bL33-R62) [[4]](diffhunk://#diff-9a449be3c254fd977bc3b6ac870fb12473832a2cd44984dd731932f64bc3967fL18-R58) [[5]](diffhunk://#diff-936a637668ee7b1e54ebac1f74f66cd5d404edba7487c5ad96ac0e98f9077d5dR1-R31) [[6]](diffhunk://#diff-42423e9218a557d905f5289fff6ab7fba4c4653d744b11cdb1d7bd0484fd72f6R1-R55) [[7]](diffhunk://#diff-2190e8acdd8d870e4539fe430467ecc497ea0d432765a6ac147f6aebd88e8741R1-R94)

### Code Enhancements:
* **Refactored local storage handling**: Improved the handling of local storage by exporting the `localStorage` instance and updating the methods to manage favorites for coins, exchanges, and wallets. (`extensions/dropstab/src/commands/dropstabCoinsCommands.ts`, `extensions/dropstab/src/commands/dropstabCoinsExchangesCommands.ts`, `extensions/dropstab/src/commands/dropstabWalletsCommands.ts`) [[1]](diffhunk://#diff-64c4f6a4ce51c1781f25726ca1b52b75754a5e7dfc58bb4fbacaade1169ccfb4L5-R5) [[2]](diffhunk://#diff-2d5802a14b43bd0b4033ef344c12295151c36a1fae6304b6adafe2cc062f40e9R1-R34) [[3]](diffhunk://#diff-2cdc571f23d0ae32448fd9372b2d6d22cfcc94afe29406ef745e9ffcbe80ae1eR4-R40)

### Dependency Updates:
* **Updated and added dependencies**: Added `lodash` and updated `micromatch` to the latest versions to improve the functionality and performance of the extension. (`extensions/dropstab/package-lock.json`, `extensions/dropstab/package.json`) [[1]](diffhunk://#diff-6854970ec30ce208d2eea1bff3b90ecede0739fbef7e611378f8942d081d706aR12) [[2]](diffhunk://#diff-6854970ec30ce208d2eea1bff3b90ecede0739fbef7e611378f8942d081d706aR1455-R1459) [[3]](diffhunk://#diff-6854970ec30ce208d2eea1bff3b90ecede0739fbef7e611378f8942d081d706aL1480-R1488) [[4]](diffhunk://#diff-9a449be3c254fd977bc3b6ac870fb12473832a2cd44984dd731932f64bc3967fL18-R58)

### Documentation:
* **Updated README and CHANGELOG**: Documented the new features and updated the local setup instructions to reflect the latest changes. (`extensions/dropstab/README.md`, `extensions/dropstab/CHANGELOG.md`) [[1]](diffhunk://#diff-fe0bcafb44232aa92569c7ed46acb031bbcd110517e51cb0761ea34e8e8a8d8bR4-R43) [[2]](diffhunk://#diff-fe0bcafb44232aa92569c7ed46acb031bbcd110517e51cb0761ea34e8e8a8d8bL33-R62) [[3]](diffhunk://#diff-b87f3c185a378b4c679ce5aedcb7810e641dda001ffe6395bc4199855afb59fbR4-R12)

### Utility Functions:
* **Implemented data fetching hooks**: Created hooks for fetching data for the gainers and losers feature, encapsulating the logic for API calls and data processing. (`extensions/dropstab/src/hooks/gainersLosersHooks.ts`)

File: extensions/dropstab/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
+## [Initial release] - 2024-08-22
+
+## [Add Ethereum Gas Tracker, Macro View, Gainers & Losers, CEX Pairs Search] - 2024-08-22
+
+### Added
+
+- Ethereum Gas Tracker
+- Macro View
+- Gainers & Losers
+- CEX Pairs Search
```
Comment (pernielsentikaer) [inline extensions/dropstab/CHANGELOG.md:12]:
```suggestion
## [Add Ethereum Gas Tracker, Macro View, Gainers & Losers, CEX Pairs Search] - {PR_MERGE_DATE}

### Added

- Ethereum Gas Tracker
- Macro View
- Gainers & Losers
- CEX Pairs Search

## [Initial release] - 2024-08-22
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Could you make this change?

**Comment by Motion-Up** [comment]
Yes, of course.

**Comment by pernielsentikaer** [comment]
Forgot that we can't change the files, could you change {PR_MERGE_DATE} to 2024-10-01

**Comment by Motion-Up** [comment]
Ready

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Comment by pernielsentikaer** [comment]
Could you make a new PR with metadata images in the right format? 

You can see here how to make [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension - please compress since we can modify in the PR

**Comment by Motion-Up** [comment]
> Could you make a new PR with metadata images in the right format?
> 
> You can see here how to make [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension - please compress since we can modify in the PR

I have created a new PR and made the updates you requested. You can review the changes here: [https://github.com/raycast/extensions/pull/14765](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html).

The metadata images have been updated to the correct format, and I have ensured they comply with the Raycast documentation guidelines. Additionally, I have compressed the images to optimize performance.

Feel free to review the PR and let me know if there are any further adjustments needed.

**Pull request merged**